src/app/components/blocks/ActivityModal.tsx

```diff
@@ -677,10 +738,16 @@ type ActivityTypeLabelProps = {
 const ActivityTypeLabel: FC<ActivityTypeLabelProps> = ({ item, className }) => {
   const isPopupMode = isPopup();
 
-  const name =
-    item.type === ActivityType.Transaction && item.source.type === "self"
-      ? "Transfer"
-      : item.type;
+  const name = useMemo(() => {
+    if (item.type === ActivityType.Transaction && item.source.type === "self") {
+      return "Transfer";
```

Comment:
Please, add also variant for "Transaction":
- item.type === ActivityType.Transaction && item.txAction?.type === TxActionType.TokenTransfer -> "Transfer"
-  item.type === ActivityType.Transaction -> "Transaction"

Reply:
1st variant just was added by myself.
2nd isn't necessary because we default case: `return item.type`